engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/component/ComponentExtractor.scala

```diff
@@ -0,0 +1,71 @@
+package pl.touk.nussknacker.engine.component
+
+import com.typesafe.config.Config
+import com.vdurmont.semver4j.Semver
+import net.ceedubs.ficus.Ficus._
+import pl.touk.nussknacker.engine.api.component._
+import pl.touk.nussknacker.engine.api.process._
+import pl.touk.nussknacker.engine.component.ComponentExtractor.componentConfigPath
+import pl.touk.nussknacker.engine.util.loader.ScalaServiceLoader
+import pl.touk.nussknacker.engine.version.BuildInfo
+
+object ComponentExtractor {
+
+  val componentConfigPath = "components"
+
+  def apply(classLoader: ClassLoader): ComponentExtractor = {
+    val version = NussknackerVersion(new Semver(BuildInfo.version))
+    ComponentExtractor(classLoader, version)
+  }
+
+}
+
+case class ComponentExtractor(classLoader: ClassLoader, nussknackerVersion: NussknackerVersion) {
+
+  private val providers = ScalaServiceLoader.load[ComponentProvider](classLoader).map(p => p.providerName -> p).toMap
```

Comment:
WDYT about making it lazy val? IMO it is not intutitive that creation of extractor already load providers

Reply:
:+1: